### PR TITLE
Show towns names on website instead of ids

### PR DESCRIPTION
**Pull request by bpawel10**

### CODE REVIEW

File: DevAAC/Models/Player.php [hunk 1]
```diff
@@ -156,23 +156,23 @@ class Player extends \Illuminate\Database\Eloquent\Model {
 
     // this is redundant
     protected $hidden = array(
-        'health', 'mana', 'manaspent', 'conditions', 'lastip', 'save', 'offlinetraining_time', 'offlinetraining_skill',
+        'health', 'mana', 'manaspent', 'town_id', 'conditions', 'lastip', 'save', 'offlinetraining_time', 'offlinetraining_skill',
```
Comment (DSpeichert) [inline DevAAC/Models/Player.php]: You don't need to hide `town_id` to add `town`.

File: DevAAC/Models/Player.php [hunk 2]
```diff
@@ -286,4 +286,15 @@ public function setLastipAttribute($longip)
     {
         $this->attributes['lastip'] = chbo(ip2long($longip));
     }
+
+    public function getTownAttribute()
+    {
+        if(defined('TOWNS'))
```
Comment (DSpeichert) [inline DevAAC/Models/Player.php]: Table towns exists in database since https://github.com/otland/forgottenserver/pull/2386

**Comment by bpawel10** [comment]
Readme says that DevAAC supports TFS 1.0+ and towns table is a part of TFS 1.3, so I'm not sure if we can use it.

**Comment by DSpeichert** [comment]
You are right. But I don't mind releasing a new version that will only support TFS 1.3+. Back when DevAAC was created TFS 1.0 was latest version.

**Comment by bpawel10** [comment]
@DSpeichert It now uses towns table, but I don't see a reason to create a new API endpoint with towns. Isn't the way it works now better?

File: DevAAC/Models/Player.php [hunk 3]
```diff
@@ -224,6 +224,16 @@ public function houseBids()
...
     }
 
+    public function town()
+    {
+        return $this->belongsTo('DevAAC\Models\Town');
+    }
+
+    public function getTownNameAttribute()
+    {
+        return $this->town->name;
```
Comment (DSpeichert) [inline DevAAC/Models/Player.php]: You should check if $this->town is not null

Comment (bpawel10) [reply DevAAC/Models/Player.php]: What for? What should I do if it's null? If I don't return anything, it'll be null anyway just like it's now.

File: DevAAC/Models/House.php [hunk 4]
```diff
@@ -75,7 +75,8 @@ public function town()
 
     public function getTownNameAttribute()
     {
-        return $this->town->name;
+        if(!is_null($this->town))
```
Comment (DSpeichert) [inline DevAAC/Models/House.php]: `return $this->town ? $this->town->name : null;`

File: DevAAC/Models/Player.php [hunk 5]
```diff
@@ -231,7 +231,8 @@ public function town()
 
     public function getTownNameAttribute()
     {
-        return $this->town->name;
+        if(!is_null($this->town))
```
Comment (DSpeichert) [inline DevAAC/Models/Player.php]: `return $this->town ? $this->town->name : null;`

**Pull request merged**